hphp/runtime/vm/jit/vasm-simplify-arm.cpp

```diff
@@ -84,17 +84,38 @@ bool simplify(Env& env, const cmovq& inst, Vlabel b, size_t i) {
 
 ///////////////////////////////////////////////////////////////////////////////
 
-bool simplify(Env& env, const loadb& inst, Vlabel b, size_t i) {
-  return if_inst<Vinstr::movzbl>(env, b, i + 1, [&] (const movzbl& mov) {
-    // loadb{s, tmp}; movzbl{tmp, d}; -> loadzbl{s, d};
-    if (!(env.use_counts[inst.d] == 1 &&
-          inst.d == mov.s)) return false;
+template<Vinstr::Opcode op>
```

Comment:
I don't see a reason for op being a template parameter instead of a function parameter. But I guess this is following the same approach used by if_inst (although I don't see why use a template there either), so I'm ok.